tests/test_cli_compile.py

```diff
@@ -2741,3 +2741,31 @@ def test_print_deprecation_warning_if_using_legacy_resolver(runner, current_reso
         assert expected_warning in out.stderr
     else:
         assert expected_warning not in out.stderr
+
+
+@pytest.mark.parametrize(
+    "input_filenames",
+    (
+        pytest.param(("requirements.txt",), id="one file"),
+        pytest.param(("requirements.txt", "dev-requirements.in"), id="multiple files"),
+    ),
+)
+def test_raise_error_when_input_and_output_filenames_are_matched(
```

Comment:
Plz add a test for `-` too.

Reply:
We have this test: https://github.com/jazzband/pip-tools/blob/23d874e5e5d82fc27d233c58627ca5bf3bef9a1f/tests/test_cli_compile.py#L1406-L1421

Reply:
Fair enough, resolving the thread.